cmd/server.go

```diff
@@ -96,7 +88,41 @@ func NewServerCommand(stopCh <-chan struct{}) *cobra.Command {
 			if err != nil {
 				logger.Fatalf("Failed to create snapshotter: %v", err)
 			}
+
+			// Start http handler for handling requests from etcd container after snapshotter
+			// has been successfully created.
+			handler := &server.HTTPHandler{
+				Port:            port,
+				EtcdInitializer: *etcdInitializer,
+				Logger:          logger,
+				Status:          http.StatusOK,
+			}
+			logger.Info("Regsitering the http request handlers...")
+			handler.RegisterHandler()
+			logger.Info("Starting the http server...")
+			go handler.Start()
+			defer handler.Stop()
 			for {
+				for {
+					if err == nil {
+						break
+					}
+					// It comes here if snapshotter fails to create a snapshot. It
+					// could be because the cloud provider was down. The assumption
+					// for this change is that the client and connection becomes in-
+					// valid and we need to re-initialize the snapshotter with a new
+					// client for the cloud provider.
+					logger.Infof("Re-initializing Snapshotter in case of error...")
+					select {
+					case <-stopCh:
+						logger.Info("Shutting down...")
+						return
+					default:
+						// TO DO: check if we need to do a exp. backoff.
```

Comment:
This will not crash process at all in case of error. And keep retrying continuously. This might work when it is used as  sidecar in statefulset since there we will get to know in case backup is failing because of liveness probe. But when used as systemd or simple command line process it won't work. So, we should definitely crash it after some retries. 
<br/>I think it should be small change to have exponential retries. Why not do it in same fix.  Something like:
```go
var exponentialRetryTimeout = 1 //(seconds)
for {
     for {
              if err== nil {
                 exponentialRetryTimeout=1
             } else {
                exponentialRetryTimeout << 1
             }
             
            if exponentialRetryTimeout > 16 { 
               fatalf("cloud error")
            } else {
                    select{
                    ...
                   }
            }
      }
      ...
}
```